### ISSUE TITLE
Student answering numerical scale question: valid values are rejected

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#10891]
Reported by a user. I was able to reproduce:

![image](https://user-images.githubusercontent.com/1673303/102802315-6a07af80-43f1-11eb-8c13-774fa6c238b3.png)

When the increment is 0.1, valid values are being rejected. I suspect the way JS handles float number precision is to blame. 
Need to fix quickly as it affects an ongoing session.

**Comment by wkurniawan07** [comment]
Java is not free from floating point errors either (I don't think any programming language is).

```
> System.out.println(1.2 / 0.1);
11.999999999999998
> System.out.println(Math.floor(1.2 / 0.1));
11.0
```

Why this didn't happen before: step validation was conveniently missed in V6
https://github.com/TEAMMATES/teammates/blob/9c60e07c260c3f4fd1b268d627f0cd4b9cda24da/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java#L724-L739

**Comment by wkurniawan07** [comment]
https://github.com/TEAMMATES/teammates/blob/4a7d83795e56cbc33f3741273debfa776bf2c7e5/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java#L48-L76
```patch
-         double interval = details.getAnswer() - minScale;
-         double remainder = convertTo5dp(interval - Math.floor(interval / step) * step);
+         BigDecimal answer = new BigDecimal(details.getAnswer());
+         BigDecimal interval = answer.subtract(new BigDecimal(minScale));
+         BigDecimal stepBd = new BigDecimal(step);
+         double remainder = interval.subtract(interval.divide(stepBd, RoundingMode.FLOOR).multiply(stepBd))
+                 .setScale(5, RoundingMode.HALF_UP)
+                 .doubleValue();
```

Someone needs to check whether this covers all cases. This is why #1501 is important.

**Comment by damithc** [comment]
Thanks for the quick investigation @wkurniawan07 
Others, please help to roll out a hot fix quickly.

**Comment by damithc** [comment]
BTW, if a proper fix is going to take a time, we can disable the step validation as an interim measure?.

**Comment by madanalogy** [comment]
Resolved with #10892
